flow-server/src/main/java/com/vaadin/flow/server/startup/RouteRegistryInitializer.java

```diff
@@ -117,6 +120,31 @@ public static void validateLayoutAnnotations(Set<Class<?>> routesSet)
             throw new InvalidRouteLayoutConfigurationException(
                     String.format(message, faultyLayoutsString));
         }
+        Map<String, List<Class<?>>> layoutsMap = new HashMap<>();
+        for (Class<?> routeClass : routesSet) {
+            if (routeClass.isAnnotationPresent(Layout.class)) {
+                String layoutValue = routeClass.getAnnotation(Layout.class)
+                        .value();
+                layoutsMap.computeIfAbsent(layoutValue, k -> new ArrayList<>());
+                layoutsMap.computeIfPresent(layoutValue, (value, classList) -> {
+                    classList.add(routeClass);
+                    return classList;
+                });
+            }
+        }
+        Set<List<Class<?>>> collect = layoutsMap.values().stream()
+                .filter(entry -> entry.size() > 1).collect(Collectors.toSet());
+        if (!collect.isEmpty()) {
+            StringBuilder messageBuilder = new StringBuilder(
+                    "Found duplicate @Layout values in classes:");
+            for (List<Class<?>> classes : collect) {
+                messageBuilder.append("\n").append(" - ")
+                        .append(classes.stream().map(clazz -> clazz.getName())
+                                .collect(Collectors.joining(" - ")));
```

Comment:
nit: a newline could make the message more readable
```suggestion
                                .collect(Collectors.joining("\n - ")));
```

Reply:
This would perhaps work if you expected there to never be more than one pair or one conflicting layout.
When there is multiple I was thinking they should all be on one line so for 2 different conflicting layouts now would give:
```
 - Class1 - Class2 - Class5
 - Class3 - Class4
 ```
Where as with a new line it would become:
``` 
 - Class 1
 - Class 2
 - Class 5
 - Class 3
 - Class 4
```
Perhaps having something noting the different conflicting sets...

Reply:
Got it. Definitely single line makes more sense